internal/connect/hwinfo.go

```diff
@@ -190,6 +183,27 @@ func uuid() (string, error) {
 	return out, nil
 }
 
+// uuidS390 returns the system uuid on S390 or "" if it cannot be found
+func uuidS390() string {
+	out, err := readValues("-u")
+	if err != nil {
+		return ""
+	}
+	uuid := string(out)
+	if isUUID(uuid) {
+		return uuid
+	}
+	Debug.Print("Not implemented. Unable to determine UUID for s390x. Set to \"\"")
+	return ""
+}
+
+// isUUID returns true if s is a valid v4 uuid
+func isUUID(s string) bool {
+	exp := `^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$`
```

Comment:
I think this regexp needs to be relaxed a bit. It's not uuid4, but RFC 4122. 
I think just `^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$` is enough